### PR TITLE
Feat/tele token

### LINKED ISSUE #12431: Create a new token when creating a telegraf config, use that token in verify step
as a telegraf user
with admin
when i create a new telegraf config
then i would like an explicit token to be created for that config
then i would like the token to have write permission to the bucket and read permission for telegraf configs
then i would like a way to connect that telegraf config to the token created

### PR DESCRIPTION
**Pull request by 121watts**
Closes #12431

- Adds telegraf configs to redux
- Creates a token of appropriate permissions for that telegraf config (read to that telegraf config / write to user specified bucket)
- Displays said token instead of the master token to the user in the "copy to CLI" step
- Stores said token as a label on the telegraf config
- Removes said label and token then telegraf config is deleted

Previously we used the master user token for telegraf permissions.  

Now, when a user creates a telegraf config we create a token with the appropriate permissions for the telegraf config.  

The way we accomplish this makes me sad.  We put a label with the token on the `TelegrafConfig`'s lables.  When the TelegrafConfig gets deleted we delete the corresponding label and token.  This makes me sad.

### CODE REVIEW

File: http/telegraf.go [hunk 1]
```diff
@@ -120,18 +121,67 @@ type telegrafLinks struct {
...
+
+	// telegrafConfigEncode is the helper struct for json encoding.
+	type telegrafConfigEncode struct {
+		ID             platform.ID                  `json:"id"`
+		OrganizationID platform.ID                  `json:"organizationID,omitempty"`
+		Name           string                       `json:"name"`
+		Description    string                       `json:"description"`
+		Agent          platform.TelegrafAgentConfig `json:"agent"`
+		Plugins        []telegrafPluginEncode       `json:"plugins"`
+		Labels         []platform.Label             `json:"labels"`
```
Comment (ischolten) [inline http/telegraf.go]: woo LABELS!

File: ui/src/authorizations/apis/index.ts [hunk 2]
```diff
@@ -0,0 +1,16 @@
+import AJAX from 'src/utils/ajax'
+
+export const createAuthorization = async authorization => {
+  try {
+    const {data} = await AJAX({
+      method: 'POST',
```
Comment (ischolten) [inline ui/src/authorizations/apis/index.ts]: should this be in the client library?

File: ui/src/authorizations/actions/index.ts [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+
+export const deleteAuthorization = (id: string, name: string = '') => async (
+  dispatch: Dispatch<Action>
+) => {
+  try {
+    await client.authorizations.delete(id)
+
+    dispatch(removeAuthorization(id))
+  } catch (e) {
+    console.log(e)
```
Comment (ischolten) [inline ui/src/authorizations/actions/index.ts]: can the console logs in this file be console.errors?

File: ui/src/configuration/components/Telegrafs.tsx [hunk 4]
```diff
@@ -0,0 +1,352 @@
...
+
+  private handleAddCollector = () => {
+    const {buckets, onSetBucketInfo, onSetDataLoadersType} = this.props
+
+    if (buckets && buckets.length) {
+      const {organization, organizationID, name, id} = buckets[0]
+      onSetBucketInfo(organization, organizationID, name, id)
+    }
+
+    onSetDataLoadersType(DataLoaderType.Scraping)
```
Comment (ischolten) [inline ui/src/configuration/components/Telegrafs.tsx]: shouldnt this type be `streaming`? or (collectors/telegraf) whatever its been named

File: ui/src/organizations/components/TelegrafInstructionsOverlay.tsx [hunk 5]
```diff
@@ -28,37 +27,54 @@ interface DispatchProps {
...
-    const {notify, collector, visible, onDismiss, username} = this.props
+    const {notify, collector, visible, onDismiss} = this.props
 
     return (
       <WizardOverlay
         visible={visible}
         title="Telegraf Setup Instructions"
         onDismiss={onDismiss}
       >
-        <FetchAuthToken username={username}>
```
Comment (ischolten) [inline ui/src/organizations/components/TelegrafInstructionsOverlay.tsx]: nice

File: ui/src/telegrafs/actions/index.ts [hunk 6]
```diff
@@ -0,0 +1,122 @@
...
+
+export const getTelegrafs = () => async (dispatch: Dispatch<Action>) => {
+  try {
+    dispatch(setTelegrafs(RemoteDataState.Loading))
+
+    const telegrafs = await client.telegrafConfigs.getAll()
+
+    dispatch(setTelegrafs(RemoteDataState.Done, telegrafs))
+  } catch (e) {
+    console.log(e)
```
Comment (ischolten) [inline ui/src/telegrafs/actions/index.ts]: can this be console.error

File: ui/src/authorizations/apis/index.ts (hunk 2 above)
Comment (121watts) [reply ui/src/authorizations/apis/index.ts]: yup.  just wanted to get this out quickly

File: ui/src/configuration/components/Telegrafs.tsx [hunk 7]
```diff
@@ -0,0 +1,352 @@
...
+    const {org} = this.props
+    const {searchTerm} = this.state
+
+    if (_.isEmpty(searchTerm)) {
+      return (
+        <EmptyState size={ComponentSize.Medium}>
+          <EmptyState.Text
+            text={`${
+              org.name
+            } does not own any Telegraf  Configurations, why not create one?`}
```
Comment (mavarius) [inline ui/src/configuration/components/Telegrafs.tsx]: There's an extra space in both of your empty space text.

File: ui/src/configuration/components/Telegrafs.tsx [hunk 8]
```diff
@@ -0,0 +1,352 @@
...
+        <EmptyState.Text text="No Telegraf  Configuration buckets match your query" />
+      </EmptyState>
+    )
+  }
+
+  private handleDeleteTelegraf = async (telegraf: Telegraf) => {
+    this.props.deleteTelegraf(telegraf.id, telegraf.name)
+
+    // hack to remove stale tokens from system when telegraf is deleted
+    const label = telegraf.labels.find(l => l.name == 'token')
```
Comment (mavarius) [inline ui/src/configuration/components/Telegrafs.tsx]: Hacky indeed.

File: ui/src/telegrafs/actions/index.ts (hunk 6 above)
Comment (mavarius) [inline ui/src/telegrafs/actions/index.ts]: Also these console logs

**Pull request merged**